core/opendaq/modulemanager/src/module_manager_impl.cpp

```diff
@@ -754,37 +755,20 @@ uint16_t ModuleManagerImpl::getServerCapabilityPriority(const ServerCapabilityPt
     return 0;
 }
 
-PropertyObjectPtr ModuleManagerImpl::populateGeneralConfig(const PropertyObjectPtr& streamingConfig)
+PropertyObjectPtr ModuleManagerImpl::populateGeneralConfig(const PropertyObjectPtr& config)
 {
-    PropertyObjectPtr resultConfig = streamingConfig.assigned() ? streamingConfig : PropertyObject();
-
-    if (!resultConfig.hasProperty("StreamingConnectionHeuristic"))
-    {
-        const auto streamingConnectionHeuristicProp =  SelectionProperty("StreamingConnectionHeuristic",
-                                                                        List<IString>("MinConnections",
-                                                                                      "MinHops",
-                                                                                      "NotConnected"),
-                                                                        0);
-        resultConfig.addProperty(streamingConnectionHeuristicProp);
-    }
-
-    if (!resultConfig.hasProperty("PrioritizedStreamingProtocols"))
-    {
-        auto prioritizedStreamingProtocols = List<IString>("opendaq_native_streaming", "opendaq_lt_streaming");
-        resultConfig.addProperty(ListProperty("PrioritizedStreamingProtocols", prioritizedStreamingProtocols));
-    }
-
-    if (!resultConfig.hasProperty("AllowedStreamingProtocols"))
-    {
-        resultConfig.addProperty(ListProperty("AllowedStreamingProtocols", List<IString>()));
-    }
+    const auto defConfig = createGeneralConfig();
+    if (!config.assigned())
+        return defConfig;
 
-    if (!resultConfig.hasProperty("AutomaticallyConnectStreaming"))
+    for (const auto& prop : defConfig.getAllProperties())
     {
-        resultConfig.addProperty(BoolProperty("AutomaticallyConnectStreaming", true));
+        const auto name = prop.getName();
+        if (config.hasProperty(name))
+            defConfig.setPropertyValue(name, config.getPropertyValue(name));
```

Comment:
can we trigger the exception is the type is mismatched? or we will get just ignore err code?